client/web/compose/src/components/Common/RecordListFilter.vue

```diff
@@ -615,7 +619,25 @@ export default {
         })
 
         return { groupCondition, filter, name }
-      }))
+      })
+    },
+
+    onSave (close = true) {
+      if (close) {
+        this.$refs.popover.$emit('close')
+      }
+
+      // Emit only value and not whole record with every filter
+      this.$emit('filter', this.processFilter())
```

Comment:
This looks like you can just have a onSave function with (close = true, type = 'filter').
So that code doesn't repeat